### PR TITLE
feat: build & publish init container

### PR DESCRIPTION
**Pull request by asarenkansah**
Creating a deployment for init container image deployment by following the example on dapr's main repo: 

https://github.com/dapr/dapr/blob/6220ab954a27b1c2e620a5060f1e056d653b3695/.github/workflows/dapr.yml#LL488C1-L489C1

### CODE REVIEW

**Comment by salaboy** [comment]
@mcruzdev you can collaborate on here..

File: .github/workflows/dapr-init-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+      ARCHIVE_OUTDIR: "dist/archives"
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target_os: linux
+            target_arch: amd64
+            job_name: "Linux/amd64"
+            sidecar_flavor: "allcomponents"
```
Comment (salaboy) [inline .github/workflows/dapr-init-workflow.yaml:36]: @asarenkansah I think this is somethign that we do only for the dapr sidecar and we don't need it for the dapr ambien init container.

File: .github/workflows/dapr-init-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+            ~/go/pkg/mod
+          key: ${{ matrix.target_os }}-${{ matrix.target_arch }}-go-${{ steps.setup-go.outputs.go-version }}-build-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ matrix.target_os }}-${{ matrix.target_arch }}-go-${{ steps.setup-go.outputs.go-version }}-build-
+      - name: Run make release to build and archive binaries
+        env:
+          GOOS: ${{ matrix.target_os }}
+          GOARCH: ${{ matrix.target_arch }}
+          ARCHIVE_OUT_DIR: ${{ env.ARCHIVE_OUTDIR }}
+        if: matrix.sidecar_flavor == 'allcomponents'
```
Comment (salaboy) [inline .github/workflows/dapr-init-workflow.yaml:112]: @asarenkansah same here..

File: .github/workflows/dapr-init-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+          DAPR_SIDECAR_FLAVOR: "${{ matrix.sidecar_flavor }}"
+        if: matrix.sidecar_flavor != 'allcomponents'
+        run: |
+          mkdir -p "${ARCHIVE_OUT_DIR}"
+          make release-flavor
+        shell: bash
+      - name: upload artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: dapr_${{ matrix.target_os }}_${{ matrix.target_arch }}_${{ matrix.sidecar_flavor }}
```
Comment (salaboy) [inline .github/workflows/dapr-init-workflow.yaml:131]: @asarenkansah we need to be careful here to not override the dapr/dapr images or artifacts.. here we should be creating the `dapr-ambient-init` container

**Comment by salaboy** [comment]
@asarenkansah ok.. great this is a good start.. I would comment the publish side of things to validate that we are creating the correct artifacts before pushing to any registry. I think we can do it step by step. We can merge a workflow to build artifacts, then when we validate that we add the publish and then the helm chart

File: .github/workflows/dapr-init-workflow.yaml (hunk 1 above)
Comment (asarenkansah) [reply .github/workflows/dapr-init-workflow.yaml:36]: Gotcha, good to know

**Pull request merged**